x-pack/plugins/apm/scripts/kibana-security/create_kibana_user_role.ts

```diff
@@ -0,0 +1,534 @@
+      },
+    },
+  });
+  await createRole({
+    roleName: KIBANA_WRITE_ROLE,
+    esPassword,
+    esUserName,
+    kibanaBaseUrl,
+    kibanaPrivileges: {
+      feature: {
+        // core
+        discover: ['all'],
+        dashboard: ['all'],
+        canvas: ['all'],
+        ml: ['all'],
+        maps: ['all'],
+        graph: ['all'],
+        visualize: ['all'],
+
+        // observability
+        logs: ['all'],
+        infrastructure: ['all'],
+        apm: ['all'],
+        uptime: ['all'],
+
+        // security
+        siem: ['all'],
+
+        // management
+        dev_tools: ['all'],
+        advancedSettings: ['all'],
+        indexPatterns: ['all'],
+        savedObjectsManagement: ['all'],
+        stackAlerts: ['all'],
+        fleet: ['all'],
+        actions: ['all'],
+      },
+    },
+  });
+
+  // read access only to APM + apm index access
+  await createOrUpdateUser({
+    newUser: {
+      username: 'apm_read_user',
+      roles: [APM_USER_ROLE, APM_READ_ROLE],
+    },
+    esPassword,
+    esUserName,
+    kibanaBaseUrl,
+  });
+
+  // read access to all apps + apm index access
+  await createOrUpdateUser({
+    newUser: {
+      username: 'kibana_read_user',
+      roles: [APM_USER_ROLE, KIBANA_READ_ROLE],
+    },
+    esPassword,
+    esUserName,
+    kibanaBaseUrl,
+  });
+
+  // read/write access to all apps + apm index access
+  await createOrUpdateUser({
+    newUser: {
+      username: 'kibana_write_user',
+      roles: [APM_USER_ROLE, KIBANA_WRITE_ROLE],
+    },
+    esPassword,
+    esUserName,
+    kibanaBaseUrl,
+  });
+}
+
+async function isSecurityEnabled({
+  esPassword,
+  esUserName,
+  kibanaBaseUrl,
+}: {
+  esPassword: string;
+  esUserName: string;
+  kibanaBaseUrl: string;
+}) {
+  try {
+    await callKibana({
+      esPassword,
+      esUserName,
+      kibanaBaseUrl,
+      options: {
+        url: `/internal/security/me`,
+      },
+    });
+    return true;
+  } catch (err) {
+    return false;
+  }
+}
+
+async function callKibana<T>({
+  esPassword,
+  esUserName,
+  kibanaBaseUrl,
+  options,
+}: {
+  esPassword: string;
+  esUserName: string;
+  kibanaBaseUrl: string;
+  options: AxiosRequestConfig;
+}): Promise<T> {
+  const kibanaBasePath = await getKibanaBasePath({ kibanaBaseUrl });
+
+  if (!esPassword) {
+    throw new Error('Missing `--password`');
+  }
+
+  const { data } = await axios.request({
+    ...options,
+    baseURL: kibanaBaseUrl + kibanaBasePath,
+    auth: {
+      username: esUserName,
+      password: esPassword,
+    },
+    headers: { 'kbn-xsrf': 'true', ...options.headers },
+  });
+  return data;
+}
+
+type Privilege = [] | ['read'] | ['all'];
+
+async function createRole({
+  esPassword,
+  esUserName,
+  kibanaBaseUrl,
+  roleName,
+  kibanaPrivileges,
+}: {
+  roleName: string;
+  kibanaPrivileges: { base?: Privilege; feature?: Record<string, Privilege> };
+  esPassword: string;
+  esUserName: string;
+  kibanaBaseUrl: string;
+}) {
```

Comment:
Can you update the signature like:

```suggestion
async function createRole({
  kibana,
  elasticsearch,
  role
}: {
  kibana: { roleSuffix: string, hostname: string}
  elasticsearch: {username: string, password: string},
  role: Role
}) {
```

`Role` should be exactly like a role is defined in https://github.com/elastic/kibana/issues/90806.
So something like:

```js
interface Role {
  elasticsearch: { cluster: ... },
  kibana: ...
}